internal/model/function_test.go

```diff
@@ -0,0 +1,221 @@
+package model_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/go-logr/logr"
+
+	"github.com/theunrepentantgeek/crddoc/internal/config"
+	"github.com/theunrepentantgeek/crddoc/internal/model"
+	"github.com/theunrepentantgeek/crddoc/internal/packageloader"
+)
+
+func TestObject_Functions_ReturnsExpectedContent(t *testing.T) {
+	t.Parallel()
+	g := NewGomegaWithT(t)
+
+	cfg := &config.Config{}
+	loader := packageloader.New(cfg, logr.Discard())
+
+	pkg, err := loader.LoadFile(testdataPath(t, "person_with_methods.go"))
+	g.Expect(err).To(Succeed())
+	g.Expect(pkg).NotTo(BeNil())
+
+	dec, ok := pkg.Declaration("PersonWithMethods")
+	g.Expect(ok).To(BeTrue())
+
+	obj, ok := dec.(*model.Object)
+	g.Expect(ok).To(BeTrue())
+
+	// Verify we have the expected functions
+	functions := obj.Functions()
+	g.Expect(functions).To(HaveLen(5))
+
+	// Check function names
+	functionNames := make([]string, len(functions))
+	for i, fn := range functions {
+		functionNames[i] = fn.Name
+	}
+	g.Expect(functionNames).To(ConsistOf("GetName", "SetName", "IsAdult", "UpdateAge", "Compare"))
+}
+
+func TestObject_Function_ReturnsExpectedFunction(t *testing.T) {
+	t.Parallel()
+	g := NewGomegaWithT(t)
+
+	cases := map[string]struct {
+		functionName   string
+		expectedExists bool
+		checkDetails   func(*testing.T, *model.Function)
+	}{
+		"GetName exists with value receiver": {
+			functionName:   "GetName",
+			expectedExists: true,
+			checkDetails: func(t *testing.T, fn *model.Function) {
+				t.Helper()
+				g := NewGomegaWithT(t)
+				g.Expect(fn.Name).To(Equal("GetName"))
+				g.Expect(fn.Receiver.Name()).To(Equal("PersonWithMethods"))
+				g.Expect(fn.IsPointerReceiver).To(BeFalse())
+				g.Expect(fn.Parameters).To(BeEmpty())
+				g.Expect(fn.Results).To(HaveLen(1))
+				g.Expect(fn.Results[0].Type.Name()).To(Equal("string"))
+			},
+		},
+		"SetName exists with pointer receiver": {
+			functionName:   "SetName",
+			expectedExists: true,
+			checkDetails: func(t *testing.T, fn *model.Function) {
+				t.Helper()
+				g := NewGomegaWithT(t)
+				g.Expect(fn.Name).To(Equal("SetName"))
+				g.Expect(fn.Receiver.Name()).To(Equal("PersonWithMethods"))
+				g.Expect(fn.IsPointerReceiver).To(BeTrue())
+				g.Expect(fn.Parameters).To(HaveLen(1))
+				g.Expect(fn.Parameters[0].Name).To(Equal("name"))
+				g.Expect(fn.Parameters[0].Type.Name()).To(Equal("string"))
+				g.Expect(fn.Results).To(BeEmpty())
+			},
+		},
+		"IsAdult exists": {
+			functionName:   "IsAdult",
+			expectedExists: true,
+			checkDetails: func(t *testing.T, fn *model.Function) {
+				t.Helper()
+				g := NewGomegaWithT(t)
+				g.Expect(fn.Name).To(Equal("IsAdult"))
+				g.Expect(fn.Results).To(HaveLen(1))
+				g.Expect(fn.Results[0].Type.Name()).To(Equal("bool"))
```

Comment:
Similarly, create a helper assertion method `assertHasResult()` that works like this:

``` go
assertHasResult(g, fn.Results, "bool")
```

Make sure both new assertions call `t.Helper()`.